website/playground/new-samples/extending-language-services/semantic-tokens-provider-example/sample.js

```diff
@@ -0,0 +1,162 @@
+                    ? getModifier(match[2].split('.').slice(1))
+                    : 0;
+
+                data.push(
+                    // translate line to deltaLine
+                    i - prevLine,
+                    // for the same line, translate start to deltaStart
+                    prevLine === i ? match.index - prevChar : match.index,
+                    match[0].length,
+                    type,
+                    modifier
+                );
+
+                prevLine = i;
+                prevChar = match.index;
+            }
+        }
+        return {
+            data: new Uint32Array(data),
+            resultId: null
+        };
+    },
+    releaseDocumentSemanticTokens: function (resultId) { }
+});
+
+// add some missing tokens
+monaco.editor.defineTheme('myCustomTheme', {
+    base: 'vs',
+    inherit: true,
+    rules: [
+        { token: 'comment', foreground: 'aaaaaa', fontStyle: 'italic' },
+        { token: 'keyword', foreground: 'ce63eb' },
+        { token: 'operator', foreground: '000000' },
+        { token: 'namespace', foreground: '66afce' },
+
+        { token: 'type', foreground: '1db010' },
+        { token: 'struct', foreground: '0000ff' },
+        { token: 'class', foreground: '0000ff', fontStyle: 'bold' },
+        { token: 'interface', foreground: '007700', fontStyle: 'bold' },
+        { token: 'enum', foreground: '0077ff', fontStyle: 'bold' },
+        { token: 'typeParameter', foreground: '1db010' },
+        { token: 'function', foreground: '94763a' },
+
+        { token: 'member', foreground: '94763a' },
+        { token: 'macro', foreground: '615a60' },
+        { token: 'variable', foreground: '3e5bbf' },
+        { token: 'parameter', foreground: '3e5bbf' },
+        { token: 'property', foreground: '3e5bbf' },
+        { token: 'label', foreground: '615a60' },
+
+        { token: 'type.static', fontStyle: 'bold' },
+        { token: 'class.static', foreground: 'ff0000', fontStyle: 'bold' }
+    ]
+});
+
+const editor = monaco.editor.create(document.getElementById("container"), {
+    value: `Available token types:
+    [comment] [string] [keyword] [number] [regexp] [operator] [namespace]
+    [type] [struct] [class] [interface] [enum] [typeParameter] [function]
+    [member] [macro] [variable] [parameter] [property] [label]
+
+Available token modifiers:
+    [type.declaration] [type.documentation] [type.member] [type.static]
+    [type.abstract] [type.deprecated] [type.modification] [type.async]
+
+Some examples:
+    [class.static.token]     [type.static.abstract]
+    [class.static.token]     [type.static]
+
+    [struct]
+
+    [function.private]
+
+An error case:
+    [notInLegend]`,
+    language: "plaintext",
+    theme: 'myCustomTheme',
+    // semantic tokens provider is disabled by default
+    'semanticHighlighting.enabled': true
+});
+
+// currently there isn't builtin token handling
+editor._themeService._knownThemes.forEach(function (theme) {
```

Comment:
I'm not sure if the samples should be suggesting to people to use internals. In any case, this code should at least reference #1833 in a comment so that it gets more publicity and hopefully gets fixed.

Reply:
Maybe this is a bit ify, but if there is not other way... People are searching for this and there isn't complete example to try and play with.

Link could be useful, but probably comment with detailed explanation would be better.

Reply:
> Link could be useful, but probably comment with detailed explanation would be better.

Well, a link is better than no link.